src/storage/checkpoint_manager.cpp

```diff
@@ -143,10 +75,18 @@ void SingleFileCheckpointWriter::CreateCheckpoint() {
 	// get the id of the first meta block
 	auto meta_block = metadata_writer->GetMetaBlockPointer();
 
-	vector<reference<SchemaCatalogEntry>> schemas;
-	// we scan the set of committed schemas
 	auto &catalog = Catalog::GetCatalog(db).Cast<DuckCatalog>();
-	catalog.ScanSchemas([&](SchemaCatalogEntry &entry) { schemas.push_back(entry); });
+
+	catalog_entry_vector_t catalog_entries;
+	D_ASSERT(catalog.IsDuckCatalog());
+
+	auto &duck_catalog = catalog.Cast<DuckCatalog>();
+	auto &dependency_manager = duck_catalog.GetDependencyManager();
+	auto &transaction_manager = DuckTransactionManager::Get(db);
+	auto dummy_transaction = CatalogTransaction(db.GetDatabase(), transaction_manager.CurrentTransactionId(),
```

Comment:
`CurrentTransactionId` and `CurrentStartTimestamp` are used here, to create a transaction for the call to GetExportOrder.
Not a huge fan but I don't see an easy way to avoid this

Reply:
Not an issue to create a dummy transaction here for reading the current state of the catalog
We did however not need the new methods, so they are removed